src/config.rs

```diff
@@ -525,14 +525,40 @@ fn default_true() -> bool {
 fn get_cached_config(repo: &str) -> Option<Result<Arc<Config>, ConfigurationError>> {
     let cache = CONFIG_CACHE.read().unwrap();
     cache.get(repo).and_then(|(config, fetch_time)| {
-        if fetch_time.elapsed() < REFRESH_EVERY {
+        if fetch_time.elapsed() < REFRESH_EVERY_SECS {
             Some(config.clone())
         } else {
             None
         }
     })
 }
 
+async fn get_fresh_config2(
+    _gh: &GithubClient,
+    repo: &Repository,
+) -> Result<Arc<Config>, ConfigurationError> {
+    let mut content_items = octocrab::instance()
+        .repos(repo.owner(), repo.name())
+        .get_content()
+        .path(CONFIG_FILE_NAME)
+        .r#ref(&repo.default_branch)
+        .send()
+        .await
+        .map_err(|e| ConfigurationError::Http(Arc::new(e.into())))?;
```

Comment:
Getting the file using the git API reduces the caching time but not completely. It looks like this way it's much shorter. Empirically (looking with `RUST_LOG=trace`) a couple of seconds, just until the connection is closed.

Reply:
I don't know yet what to think of `REFRESH_EVERY_SECS`.